skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -460,66 +446,60 @@ public function executeEditPage() {
 		}
 
 		$this->displaytitle = true;
-		$this->title = wfMsg( $titleMsg, htmlspecialchars( $wgTitle->getPrefixedText() ) );
+		$this->title = wfMessage( $titleMsg, $this->wg->Title->getPrefixedText() )->parse();
 
 		// back to article link
 		if ( !$isPreview && !$isShowChanges ) {
-			$this->subtitle = Wikia::link( $wgTitle, wfMsg( 'oasis-page-header-back-to-article' ), [ 'accesskey' => 'c' ], [ ], 'known' );
+			$this->subtitle = Linker::linkKnown( $this->wg->Title, wfMessage( 'oasis-page-header-back-to-article' )->escaped(), [ 'accesskey' => 'c' ] );
 		}
 
 		// add edit button
 		if ( $isDiff || ( $isHistory && !$this->isUserTalkArchiveModeEnabled ) ) {
 			$this->prepareActionButton();
 
 			// show only "My Tools" dropdown on toolbar
-			$wgShowMyToolsOnly = true;
+			$this->wg->ShowMyToolsOnly = true;
 		}
 
 		// render edit dropdown / commments chicklet on history pages
 		if ( $isHistory ) {
 			// FB#1137 - re-add missing log and undelete links
 			$logPage = SpecialPage::getTitleFor( 'Log' );
-			$this->subtitle .= ' | ' . Wikia::link(
+			$this->subtitle .= wfMessage( 'pipe-separator' )->escaped() . Linker::linkKnown(
 					$logPage,
-					wfMsgHtml( 'viewpagelogs' ),
+					wfMessage( 'viewpagelogs' )->escaped(),
 					[ ],
-					[ 'page' => $wgTitle->getPrefixedText() ],
-					[ 'known', 'noclasses' ]
+					[ 'page' => $this->wg->Title->getPrefixedText() ]
 				);
 
-			// FIXME: Skin is now an abstract class (MW1.19)
-			// (wladek) created non-abstract FakeSkin class, is it the correct solution?
-			$sk = new FakeSkin();
-			$sk->setRelevantTitle( $wgTitle );
-
-			$undeleteLink = $sk->getUndeleteLink();
+			$undeleteLink = $this->getContext()->getSkin()->getUndeleteLink();
 
 			if ( !empty( $undeleteLink ) ) {
-				$this->subtitle .= ' | ' . $undeleteLink;
+				$this->subtitle .= wfMessage( 'pipe-separator' )->escaped() . $undeleteLink;
 			}
 
 			// dropdown actions
 			$this->dropdown = $this->getDropdownActions();
 
 			// use service to get data
-			$service = new PageStatsService( $wgTitle->getArticleId() );
+			$service = new PageStatsService( $this->wg->Title->getArticleID() );
 
 			// comments
 			$this->comments = $service->getCommentsCount();
 		}
 
-		wfRunHooks( 'PageHeaderEditPage', [ &$this, $ns, $isPreview, $isShowChanges, $isDiff, $isEdit, $isHistory ] );
+		Hooks::run( 'PageHeaderEditPage', [ &$this, $this->wg->Title->getNamespace(), $isPreview, $isShowChanges, $isDiff, $isEdit, $isHistory ] );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)